src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -61,11 +61,16 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 return null;
             }
 
-            var tempDataDictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
             var session = context.Session;
+            if (session == null)
```

Comment:
Should we additionally have a check for this as part of setting the temp data?
So

`TempData["foo"] = value;` would throw if the backing TempDataProvider isn't "enabled". It avoids user exceptions being lost to ether \ logging.